### ISSUE TITLE
When multiple services use the same method names, client types are not properly generated

### ISSUE THREAD

**Issue by johnholiver** [goadesign/goa#2391]
I'm facing an issue that is fully described [here](https://stackoverflow.com/questions/58885742/client-fails-to-build-in-goa) by someone else with example code.

I have a design that contains more than one `Service` definitions in which the `Method` "x" appears in each. Generation of client types will work on one service out of the multiple, but all the other services fail to generate some requestbody types (from code,  other types too).

I've noticed that there's a difference in the implementation of these two files:
- `/http/codegen/server_types.go`
- `/http/codegen/client_types.go`

While in server types it doesn't seem to matter if a requestbody was already `seen`, in client_types .go:[72,92,118,etc] there's a check to see if a type has been `seen`.
Snippet:
```
if _, ok := seen[data.Name]; ok {
    continue
}
```

Considering that server_types.go don't make use of the `seen` variable (described as `seen keeps track of the names of the types that have already been generated to prevent duplicate code generation`) **is it possible that some refactor happened at some point that removed the need for this tracking in server_types and should be matched in client_types?**

**Comment by johnholiver** [comment]
@raphael since I was looking to find any changes in this issue today, I've took the time to look into the github page and annotate the files. 

My original assumption that the condition was **removed** from the server_types.go seems to be incorrect. It appears that these `if seen` conditions were added by you last month with https://github.com/goadesign/goa/pull/2346.

Now, I was trying to setup a example so I could copy here, but it seems that this was fixed in v3.0.8. To the best of your knowledge, do you confirm that this issue was known and it got fixed?

Below there's a small design in case you need one for testing.
```
package design

import (
    . "goa.design/goa/v3/dsl"
)

var _ = API("showex", func() {
})

var _ = Service("srvc1", func() {

    HTTP(func() {
        Path("/srvc1")
    })

    Method("show", func() {

        Payload(func() {
            Attribute("id", String)
            Required("id")
        })

        Result(Srvc1Result)

        HTTP(func() {
            GET("/{id}")
            Response(StatusOK)
        })
    })

})

var _ = Service("srvc2", func() {

    HTTP(func() {
        Path("/srvc2")
    })

    Method("show", func() {

        Payload(func() {
            Attribute("id", String)
            Required("id")
        })

        Result(Srvc2Result)

        HTTP(func() {
            GET("/{id}")
            Response(StatusOK)
        })
    })

})

var Srvc1Result = ResultType("application/vnd.xxx.srvc1", func() {
    Attributes(func() {
        Attribute("id", String)
    })
})

var Srvc2Result = ResultType("application/vnd.xxx.srvc2", func() {
    Attributes(func() {
        Attribute("id", String)
    })
})
```

Result from v3.0.7
```
$go build ./cmd/showex
$ go build ./cmd/showex-cli
# goa.test/test/gen/http/srvc2/client
gen/http/srvc2/client/encode_decode.go:67:10: undefined: ShowResponseBody
gen/http/srvc2/client/types.go:16:30: undefined: ShowResponseBody
```

**Comment by raphael** [comment]
Yes you're correct that this issue was fixed in #2346. Apologies for not updating this issue and thank you for pointing this out! Closing this issue.
